### PR TITLE
Fixes milk

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Skeleton, plasmamen, and bone golems now heal from milk and get harmed by bone hurting juice, as they should.

They also now properly OD on bone hurting juice and take stamina damage from it.

Also, bone hurting juice now makes you say "oof ouch" in on_mob_add(), not on_mob_metabolize(). This means that people without livers who drink it should still say "oof ouch" upon doing so, which I think is fine.

It also slightly edits some of the messages from bone hurting juice (it replaces "oofs softly" with "oofs silently" so that my immersion isn't broken after a mime chugs some bone hurting juice and replaces "oof!" with "oof." to keep with the deadpan style of the original meme).

## Why It's Good For The Game

Y'all didn't merge my previous PR that attempted to fix this (https://github.com/tgstation/tgstation/pull/46468) because it didn't fix the underlying issue. Since this PR DOES fix the underlying issue, can we please make healing yourself as a skeleton not really, really annoying by merging this PR quickly?

## Changelog
:cl: ATHATH
fix: the three skeleton-like races now get healed by milk properly (and get damaged by and overdose on bone hurting juice properly)
tweak: makes a few minor edits to some of the messages from bone hurting juice
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
travis why

**Comment by ExcessiveUseOfCobblestone** [comment]
fix handle chemicals, stop trying to put it off species

**Comment by ATH1909** [comment]
> fix handle chemicals, stop trying to put it off species

I *did* fix handle_chemicals! The calcium healer race processing for milk is even still in there!

**Comment by kriskog** [comment]
Why plasmamen? They are made from plasma, not calcium. If anything, drinking liquid plasma should heal them not milk.
Besides, they process healing chems and you can even use bruise packs and ointment on them. They don't need this.

**Comment by PKPenguin321** [comment]
@kriskog its a fix, not a balance change, so make a separate PR for it

**Comment by kriskog** [comment]
Uhuh, I guess ¯\\\_(ツ)\_/¯

**Comment by wesoda25** [comment]
You fool, you absolute buffoon, you really thought fixing milk would be this easy?

**Comment by ATH1909** [comment]
@ExcessiveUseOfCobblestone there, NOW are you happy?

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -990,12 +990,45 @@
...
+				var/selected_part
+				switch(rand(1, 4)) //God help you if the same limb gets picked twice quickly.
+					if(1)
+						selected_part = BODY_ZONE_L_ARM
+					if(2)
+						selected_part = BODY_ZONE_R_ARM
+					if(3)
+						selected_part = BODY_ZONE_L_LEG
+					if(4)
+						selected_part = BODY_ZONE_R_LEG
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]:
```DM
var/selected_part = pick(a,b,c,d)
```
cringe

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 2]
```diff
@@ -990,12 +990,45 @@
...
+				if(1)
+					var/list/possible_says = list("oof.", "ouch!", "my bones.", "oof ouch.", "oof ouch my bones.")
+					H.say(pick(possible_says), forced = /datum/reagent/toxin/bonehurtingjuice)
+				if(2)
+					var/list/possible_mes = list("oofs softly.", "looks like their bones hurt.", "grimaces, as though their bones hurt.")
+					H.say("*custom " + pick(possible_mes), forced = /datum/reagent/toxin/bonehurtingjuice)
+				if(3)
+					to_chat(H, "<span class='warning'>Your bones hurt!</span>")
+		if(chem.volume >= chem.overdose_threshold)
+			chem.overdosed = TRUE
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: I was thinking putting this where the check_reagent proc is called, not have to manually do it per each reagent.

For handle_chemicals all you should have to do is check to see if `chem.overdosed` is T

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 3]
```diff
@@ -990,12 +990,45 @@
...
-
 	if(chem.type == /datum/reagent/toxin/bonehurtingjuice)
+		H.adjustStaminaLoss(7.5, 0)
 		H.adjustBruteLoss(0.5, 0)
-		H.reagents.remove_reagent(chem.type, REAGENTS_METABOLISM)
+		if(prob(20))
+			switch(rand(1, 3))
+				if(1)
+					var/list/possible_says = list("oof.", "ouch!", "my bones.", "oof ouch.", "oof ouch my bones.")
+					H.say(pick(possible_says), forced = /datum/reagent/toxin/bonehurtingjuice)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: `H.say("[pick(each individual item here, not prebuilt in list)]", ....)`

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 4]
```diff
@@ -990,12 +990,45 @@
...
+				if(2)
+					var/list/possible_mes = list("oofs softly.", "looks like their bones hurt.", "grimaces, as though their bones hurt.")
+					H.say("*custom " + pick(possible_mes), forced = /datum/reagent/toxin/bonehurtingjuice)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/species_types/golems.dm]: use
https://github.com/tgstation/tgstation/blob/9e35e314ab91c2b583f4fe5fcd107f4ed18885e0/code/modules/mob/emote.dm#L2

File: code/modules/mob/living/carbon/human/species_types/golems.dm (hunk 3 above)
Comment (PKPenguin321) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: doesn't really matter, does it?

**Comment by ATH1909** [comment]
@ExcessiveUseOfCobblestone imagine delaying the merging of a fix to a problem that breaks a race an a type of side-antag (pirates) because I didn't tidy up the inner workings/behind the scenes formatting of the code when I ported it over

I still followed/complied with your suggestions, but god was this painful and unfun for me to do (I had to learn emote code and copy+paste all of my changes 3 times each)

can we FINALLY get this over with now?

**Comment by ATH1909** [comment]
I mean, I do get WHY you made your suggestions and I do appreciate them, but I needed SOMEONE to vent at, and you were the easiest person to do that to, cobby.

sorry

**Comment by davethwave** [comment]
We also have Halloween races on right now so might want to hurry.

File: code/modules/mob/living/carbon/human/species_types/golems.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/species_types/golems.dm]: saves a var, pretty minor

**Comment by ExcessiveUseOfCobblestone** [comment]
Imagine thinking I won't delay your PR because code needs to be made right and you fell for the sunk costs 😏

**Comment by PKPenguin321** [comment]
Very based

**Comment by pireamaineach** [comment]
Gamer moment. Time to make plaamamen unable to metabolise reagents.

**Pull request merged**